src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -237,8 +234,78 @@ private async Task UpdateJobRecordAsync(CancellationToken cancellationToken)
                 queryParametersList.Add(Tuple.Create(KnownQueryParameterNames.Type, _exportJobRecord.ResourceType));
             }
 
+            var filteredResources = new HashSet<string>();
+            if (_exportJobRecord.Filters != null)
+            {
+                foreach (var filter in _exportJobRecord.Filters)
+                {
+                    filteredResources.Add(filter.ResourceType);
+                }
+            }
+
             IAnonymizer anonymizer = IsAnonymizedExportJob() ? await CreateAnonymizerAsync(cancellationToken) : null;
 
+            if (!progress.RunFilteredSearches)
+            {
+                var index = -1;
+                if (_exportJobRecord.Filters != null)
+                {
+                    foreach (var filter in _exportJobRecord.Filters)
+                    {
+                        index++;
+
+                        if (progress.CurrentFilter != null && filter != progress.CurrentFilter)
+                        {
+                            continue;
+                        }
+
+                        if (_exportJobRecord.ExportType == ExportJobType.All || filter.ResourceType.Equals(KnownResourceTypes.Patient, StringComparison.Ordinal))
+                        {
+                            progress.SetFilter(filter);
+
+                            List<Tuple<string, string>> filterQueryParametersList = new List<Tuple<string, string>>(queryParametersList);
+                            foreach (var param in filter.Parameters)
+                            {
+                                filterQueryParametersList.Add(param);
+                            }
+
+                            await SearchWithFilter(exportJobConfiguration, progress, filter.ResourceType, filterQueryParametersList, sharedQueryParametersList, anonymizer, "filter" + index + "-", cancellationToken);
```

Comment:
Similarly, here also we are including `_type` as part of the queryparameters (if _exportJobRecord.ResourceType != null) and passing a higher level resource type filter in `filter.ResourceType`.

Reply:
Here I think it is needed. By my reading of the spec if the user enters the _type parameter and _typeFilter parameter the _type can override the _typeFiler. So if they have `_type=Observation&_typeFilter=MedicationRequest%3Fstatus%3Dactive` no MedicationRequests will be returned because the _type parameter isn't set to return any.

Reply:
I agree that `_type` overrides whatever is in `_typefilter`. But I also feel that ideally we shouldn't execute the above query since it's a no-op.